src/map-feature.js

```diff
@@ -0,0 +1,266 @@
+export class MapFeature extends HTMLElement {
+    static get observedAttributes() {
+      return ['zoom'];
+    }
+
+    get zoom() {
+      return +(this.hasAttribute("zoom") ? this.getAttribute("zoom") : 0);
+    }
+
+    set zoom(val) {
+      var parsedVal = String.toString(parseInt(val,10));
+      if (!isNaN(parsedVal) && (parsedVal >= 0 && parsedVal <= 25)) {
+        this.setAttribute('zoom', parsedVal);
+      }
+    }
+
+    get extent() {
+      if (this.isConnected) {
+        return this._getFeatureExtent();
+      }
+    }
+
+    attributeChangedCallback(name, oldValue, newValue) {
+      switch (name) {
+        case 'zoom': {
+          if (oldValue !== newValue && this._layerParent) {
+            this._remove();
+            this._redraw();
+          }
+          break;
+        }
+      }
+    }
+
+    constructor() {
+      // Always call super first in constructor
+      super();
+    }
+
+    connectedCallback() {
+      if (this.parentNode.nodeType !== document.DOCUMENT_FRAGMENT_NODE && 
+          this.parentNode.tagName.toLowerCase() !== 'layer-') {
+        return;
+      }
+
+      this._layerParent = this.parentNode._layer ? this.parentNode._layer : this.parentNode.host._layer;
```

Comment:
I think I understand but a comment may be warranted: the feature can be a child of `<layer->`, directly, or may have been created by migration from a `<mapml->` document loaded via `<layer- src=...>`, in which case it is re-parented to exist in the shadowRoot of the `<layer- src>.shadowRoot` (a document fragment).  Is that right?